File: core/src/com/g3ida/withflyingcolours/core/player/animation/PlayerAnimationSystem.java
package com.g3ida.withflyingcolours.core.player.animation;

import com.artemis.ComponentMapper;
import com.artemis.annotations.All;
import com.artemis.systems.IteratingSystem;

import games.rednblack.editor.renderer.components.TransformComponent;

@All({PlayerAnimationComponent.class, TransformComponent.class})
public class PlayerAnimationSystem extends IteratingSystem {
    private ComponentMapper<PlayerAnimationComponent> mPlayerAnimationComponent;
    private ComponentMapper<TransformComponent> mTransformComponent;

    public PlayerAnimationSystem() {
        super();
    }

    @Override
    protected void process(int entityId) {
        PlayerAnimationComponent playerAnimation = mPlayerAnimationComponent.get(entityId);
        TransformComponent transform = mTransformComponent.get(entityId);

        if (playerAnimation.doSqueeze) {
            playerAnimation.doSqueeze = false;
            playerAnimation.squeezeAnimation.start();
        }
        else if (playerAnimation.doScale) {
            playerAnimation.doScale = false;
            playerAnimation.scaleAnimation.start();
        }

        playerAnimation.scaleAnimation.step(transform, world.delta);
        playerAnimation.squeezeAnimation.step(transform, world.delta);
    }
}

File: core/src/com/g3ida/withflyingcolours/core/player/animation/PlayerAnimationSystem.kt
package com.g3ida.withflyingcolours.core.player.animation

import com.artemis.ComponentMapper
import com.artemis.systems.IteratingSystem
import games.rednblack.editor.renderer.components.TransformComponent
import com.artemis.annotations.All

@All(PlayerAnimationComponent::class, TransformComponent::class)
class PlayerAnimationSystem : IteratingSystem() {
    private lateinit var mPlayerAnimationComponent: ComponentMapper<PlayerAnimationComponent>
    private lateinit var mTransformComponent: ComponentMapper<TransformComponent>
    override fun process(entityId: Int) {
        val playerAnimation = mPlayerAnimationComponent[entityId]
        val transform = mTransformComponent[entityId]

        if (playerAnimation.currentAnimation.isDone())
            playerAnimation.reset()
        else if (!playerAnimation.currentAnimation.isRunning())
            playerAnimation.currentAnimation.start()

        playerAnimation.currentAnimation.step(transform, world.delta)
    }
}